### PR TITLE
Refactor ActiveModel::ErrorSerializer

### PR DESCRIPTION
**Pull request by mamhoff**
## What is the current behavior?

No behavior change, but the ActiveModel::ErrorSerializer reads much nicer.

## Checklist

Please make sure the following requirements are complete:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes /
  features)
- [ ] All automated checks pass (CI/CD)

### CODE REVIEW

File: lib/jsonapi/active_model_error_serializer.rb [hunk 1]
```diff
@@ -12,35 +12,15 @@ class ActiveModelErrorSerializer < ErrorSerializer
...
-      end
-
-      errors_object.full_message(error_key, message)
+    attribute :detail do |object, _params|
+      object.full_message
     end
 
     attribute :source do |object, params|
-      error_key, _ = object
+      error_key = object.attribute
```
Comment (lilasquared) [inline lib/jsonapi/active_model_error_serializer.rb:23]: it seems before this change an array was supported here as `object` but now it is not.  Is there a reason for the change in behavior there?

Comment (mamhoff) [reply lib/jsonapi/active_model_error_serializer.rb:23]: No, that was certainly not intended, but there was also no test for it. If you depend on that behaviour, as a spec and an `Array.wrap` or similar as a PR.

Comment (lilasquared) [reply lib/jsonapi/active_model_error_serializer.rb:23]: if this is more correct i can update on my end.  we were building up these errors manually (for some unknown reason) which this suddenly broke.

**Comment by mamhoff** [comment]
I'm not the maintainer, just did this particular refactoring. So that
decision would be on you and @stas.

Aaron Roberts ***@***.***> schrieb am Mo., 15. Juli 2024,
19:26:

**Pull request merged**